xarray/tests/test_dataset.py

```diff
@@ -4035,6 +4035,19 @@ def data_set(seed=None):
     return create_test_data(seed)
 
 
+def test_constructor_aligns_to_explicit_coords():
+
+    coords = {'x': range(3)}
+    a = xr.DataArray([1, 2, 3], dims=['x'], coords={'x': range(2, -1, -1)})
```

Comment:
It would be good to make sure this also works when `coords` is passed in other forms that are valid for xarray, e.g., a tuple `(dims, values)` or `xarray.Variable` objects.